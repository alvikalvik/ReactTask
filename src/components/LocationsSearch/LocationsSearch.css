.locations-search {
  flex-shrink: 0;
  flex-grow: 1;
  margin-right: 0.2rem;
  max-width: 45rem;
  border-top: 1px solid var(--primary-color);
  text-align: center;
}

.locations-search__title {
  margin: 1rem 0 0 0;
  padding: 0;
}

.locations-search__input {
  margin-top: 0.5rem;
}

.locations-search__input-description {
  margin-top: 0.2rem;
  font-size: var(--fz-small);
}
