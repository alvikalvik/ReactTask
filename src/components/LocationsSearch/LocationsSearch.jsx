import PropTypes from 'prop-types';
import { useState } from 'react';
import { CurrentLocationInfoType } from '../../types/types';
import Preloader from '../Preloader/Preloader';
import './LocationsSearch.css';
import LocationsSearchReuslts from './LocationsSearchReuslts/LocationsSearchReuslts';
import debounce from 'lodash/debounce';
import { DEBOUNCE_WAIT_TIME, MIN_CHARACTERS_FOR_SEARCH } from '../../constants/constants';

function LocationsSearch({ isReadyForSearch, searchResults, clearSearch, searchLocations }) {
  const [inputValue, setInputValue] = useState('');
  const debouncedSearchLocations = debounce(searchLocations, DEBOUNCE_WAIT_TIME);

  function handleInputChange(event) {
    const value = event.target.value;
    setInputValue(value);

    if (value === '') {
      clearSearch();
    }

    if (value.length >= MIN_CHARACTERS_FOR_SEARCH) {
      debouncedSearchLocations(value);
    }
  }

  return (
    <div className="locations-search">
      <h2 className="locations-search__title">Search for location</h2>
      <input
        type="text"
        className="locations-search__input"
        placeholder="search query"
        value={inputValue}
        onChange={handleInputChange}
      />
      <div className="locations-search__input-description">
        at least {MIN_CHARACTERS_FOR_SEARCH} characters
      </div>
      {isReadyForSearch ? <LocationsSearchReuslts searchResults={searchResults} /> : <Preloader />}
    </div>
  );
}

LocationsSearch.propTypes = {
  isReadyForSearch: PropTypes.bool.isRequired,
  searchResults: PropTypes.arrayOf(CurrentLocationInfoType).isRequired,
  clearSearch: PropTypes.func.isRequired,
  searchLocations: PropTypes.func.isRequired
};

export default LocationsSearch;
