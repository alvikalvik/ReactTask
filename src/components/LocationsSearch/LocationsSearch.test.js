import Enzyme, { mount, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import LocationsSearch from './LocationsSearch';

Enzyme.configure({ adapter: new Adapter() });

describe('<LocationsSearch />', () => {
  it('should render a LocationsSearch with LocationsSearchReuslts', () => {
    const wrapper = shallow(
      <LocationsSearch
        isReadyForSearch={true}
        searchResults={[]}
        clearSearch={() => {}}
        searchLocations={() => {}}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('should render a LocationsSearch with Preloader', () => {
    const wrapper = shallow(
      <LocationsSearch
        isReadyForSearch={false}
        searchResults={[]}
        clearSearch={() => {}}
        searchLocations={() => {}}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
