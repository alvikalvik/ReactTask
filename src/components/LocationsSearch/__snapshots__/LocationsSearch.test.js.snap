// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LocationsSearch /> should render a LocationsSearch with LocationsSearchReuslts 1`] = `
<div
  className="locations-search"
>
  <h2
    className="locations-search__title"
  >
    Search for location
  </h2>
  <input
    className="locations-search__input"
    onChange={[Function]}
    placeholder="search query"
    type="text"
    value=""
  />
  <div
    className="locations-search__input-description"
  >
    at least 
    3
     characters
  </div>
  <LocationsSearchReuslts
    searchResults={Array []}
  />
</div>
`;

exports[`<LocationsSearch /> should render a LocationsSearch with Preloader 1`] = `
<div
  className="locations-search"
>
  <h2
    className="locations-search__title"
  >
    Search for location
  </h2>
  <input
    className="locations-search__input"
    onChange={[Function]}
    placeholder="search query"
    type="text"
    value=""
  />
  <div
    className="locations-search__input-description"
  >
    at least 
    3
     characters
  </div>
  <Preloader />
</div>
`;
